### DISCUSSION TITLE
Error when dynamically obtaining table name TS type

### DISCUSSION THREAD

**Discussion by wjw020206** [prisma/prisma#21554] [Q&A] [answered]
### Question

I try to write a method to determine whether the data exists in the data table by passing in parameters and the table name.
But when I use `prisma[table]` a type error occurs.The specific code is as follows.
```ts
import { PrismaClient } from '@prisma/client';
import { ValidationOptions, registerDecorator } from 'class-validator';

export function IsNotExists(
  table: string,
  fields: string[],
  validationOptions?: ValidationOptions
) {
  return function (object: Record<string, any>, propertyName: string) {
    registerDecorator({
      name: 'IsNotExists',
      target: object.constructor,
      propertyName: propertyName,
      constraints: [table],
      options: validationOptions,
      validator: {
        async validate(value: string) {
          const prisma = new PrismaClient({ log: ['query'] });
          const res = await prisma[table].findFirst({
            where: {
              OR: fields.map(filed => ({ [filed]: value }))
            }
          });
          return !Boolean(res);
        }
      }
    });
  };
}
```

Error message
```
error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'PrismaClient<{ log: "query"[]; }, never, DefaultArgs>'. 
No index signature with a parameter of type 'string' was found on type 'PrismaClient<{ log: "query"[]; }, never, DefaultArgs>'.
```

![image](https://github.com/prisma/prisma/assets/59106659/0d2c1067-295a-4648-85c4-674bb922097f)

### Expected behavior (optional)

The type does not report an error and can be queried normally.

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: Windows
- Database: MySQL 8.0.34
- Node.js version: 18.18.2
- Prisma: 5.4.2

**Comment by tetthys** [comment]
I am in a simillar situation. Have you already resolved this???

**Reply by wjw020206** [reply]
I didn't solve this problem correctly, I used `any` to ignore the error.
Wrote `prisma[table]` as `prisma[table as any] as any`.
```ts
import { PrismaClient } from '@prisma/client';
import { ValidationOptions, registerDecorator } from 'class-validator';

export function IsExists(
  table: string,
  fields: string[],
  validationOptions?: ValidationOptions
) {
  return function (object: Record<string, any>, propertyName: string) {
    registerDecorator({
      name: 'IsExists',
      target: object.constructor,
      propertyName: propertyName,
      constraints: [table],
      options: validationOptions,
      validator: {
        async validate(value: string) {
          const prisma = new PrismaClient({ log: ['query'] });
          const res = await (prisma[table as any] as any).findFirst({
            where: {
              OR: fields.map(filed => ({ [filed]: value }))
            }
          });
          return Boolean(res);
        }
      }
    });
  };
}

```

**Reply by tetthys** [reply, marked as answer]
(UPDATED)

Thank you!

Your idea helped me a lot.

I am not sure that this is really the only correct way.

But I suggest you better one!

For example, Let's assume there is countOf function.

```
const countOf = async (modelName: Uncapitalize<Prisma.ModelName>) => {
  const count: number = await (prisma[modelName] as any).count();
  return count;
};
```

I looked up a lot of information about this, but in the end it gets very complicated if you don't use ```any``` .

**Reply by ludralph** [reply]
Hi @wjw020206 👋 

do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
